content/agent/faq/agent-5-custom-agent-check.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Custom check, Agent v5 example
+kind: faq
+---
+
+## An HTTP Check
+
+Let's write a basic check that checks the status of an HTTP endpoint. On each run of the check, a *GET* request is made to the HTTP endpoint. Based on the response, one of the following happens:
+
+* If the response is successful (response is 200, no timeout) the response time is submitted as a metric.
+* If the response times out, an event is submitted with the URL and timeout.
+* If the response code != 200, an event is submitted with the URL and the response code.
+
+### Configuration
+
+First let's define how the configuration should look so that we know
+how to handle the structure of the `instance` payload that is passed into the
+call to `check`.
+
+Besides just defining a URL per call, it'd be nice to allow you to set a timeout for each URL. We'd also want to be able to configure a default timeout if no timeout value is given for a particular URL.
+
+So our final configuration looks something like this:
+
+```yaml
+init_config:
+    default_timeout: 5
+
+instances:
+    -   url: https://google.com
+
+    -   url: http://httpbin.org/delay/10
+        timeout: 8
+
+    -   url: http://httpbin.org/status/400
+
+```
+
+### The Check
+
+Now let's define our check method. The main part of the check makes
+a request to the URL and time the response time, handling error cases as it goes.
+
+In this snippet, we start a timer, make the GET request using the
+[requests library](http://docs.python-requests.org/en/latest/) and handle and
+errors that might arise.
+
+```python
+
```

Comment:
Could you please remove this extra newline?

Reply:
👌